sdk/js/src/ConfigRequestConsolidator.ts

```diff
@@ -0,0 +1,67 @@
+import { BucketedUserConfig } from '@devcycle/types'
+import { DVCPopulatedUser } from './User'
+
+type PromiseResolver = {
+    resolve: (value: BucketedUserConfig) => void
+    reject: (err?: any) => void
+}
+
+/**
+ * Ensures we only have one active config request at a time
+ * any calls made while another is ongoing will be merged together by using the latest user object provided
+ */
+export class ConfigRequestConsolidator {
+    currentPromise: Promise<BucketedUserConfig> | null
+    resolvers: PromiseResolver[] = []
+
+    constructor(
+        private requestConfigFunction: (user: DVCPopulatedUser) => Promise<BucketedUserConfig>,
+        private handleConfigReceivedFunction: (user: DVCPopulatedUser, config: BucketedUserConfig) => void,
+        private nextUser: DVCPopulatedUser
+    ) {}
+
+    async queue(user?: DVCPopulatedUser): Promise<BucketedUserConfig> {
+        if (user) {
+            this.nextUser = user
+        }
+
+        const resolver = new Promise<BucketedUserConfig>((resolve, reject) => {
+            this.resolvers.push({
+                resolve, reject
+            })
+        })
+
+        if (!this.currentPromise) {
+            this.processQueue()
+        }
+
+        return resolver
+    }
+
+    private async processQueue() {
+        if (!this.resolvers.length) {
+            return
+        }
+
+        const resolvers = this.resolvers.splice(0)
+        const bucketedConfig = await this.performRequest(this.nextUser).then((result) => {
```

Comment:
do you need to return `result` somewhere in this callback?

Reply:
yeah in order for the return value to be correct, you're right.
 However I'm not actually using the return value of this method so I might just have it return nothing

Reply:
by callback I mean the `.then`, im just not entirely sure how the bucketedConfig variable is going to be set, its kinda making my head spin

Reply:
oh I see what youre saying, that makes sense